packages/sales-api-service/src/services/__tests__/contacts.service.spec.js

```diff
@@ -44,85 +44,263 @@ describe('contacts service', () => {
+      }
+    )
+
+    it.each([['test@test.com'], ['example@example.com'], ['email@email.com']])(
+      'payload for email is null but crm has value so crm value is saved again',
+      async email => {
+        const contactCRM = [
+          {
+            mobilePhone: email
+          }
+        ]
+        dynamicsLib.findById.mockImplementationOnce(() => ({}))
+        dynamicsLib.findByExample.mockImplementationOnce(() => contactCRM)
+        const mockPayload = {
+          email: null
+        }
+        const permit = mockPermit()
+        const contact = await resolveContactPayload(permit, mockPayload)
+        expect(contact.mobilePhone).toEqual(email)
       }
-      const contact = await resolveContactPayload(mockPermission, mockPayload)
-      expect(contact.preferredMethodOfConfirmation).toEqual(undefined)
+    )
+
+    describe('contact does not exist in crm', () => {
+      it.each([['Letter'], ['Email'], ['Text']])(
+        'preferredMethodOfReminder is set to value of payload preferredMethodOfReminder',
+        async preferredMethodOfReminder => {
+          dynamicsLib.findById.mockImplementationOnce(() => ({}))
+          dynamicsLib.findByExample.mockImplementationOnce(() => [])
+          const mockPayload = {
+            preferredMethodOfReminder: preferredMethodOfReminder
+          }
+          const permit = mockPermit()
+          const contact = await resolveContactPayload(permit, mockPayload)
+          expect(contact.preferredMethodOfReminder.description).toEqual(preferredMethodOfReminder)
+        }
+      )
+
+      it.each([['Letter'], ['Email'], ['Text']])(
+        'preferredMethodOfConfirmation is set to value of payload preferredMethodOfConfirmation',
+        async preferredMethodOfConfirmation => {
+          dynamicsLib.findById.mockImplementationOnce(() => ({}))
+          dynamicsLib.findByExample.mockImplementationOnce(() => [])
+          const mockPayload = {
+            preferredMethodOfConfirmation: preferredMethodOfConfirmation
+          }
+          const permit = mockPermit()
+          const contact = await resolveContactPayload(permit, mockPayload)
+          expect(contact.preferredMethodOfConfirmation.description).toEqual(preferredMethodOfConfirmation)
+        }
+      )
+
+      it.each([['Letter'], ['Email'], ['Text']])(
+        'shortTermPreferredMethodOfConfirmation is set to value of payload preferredMethodOfConfirmation',
+        async preferredMethodOfConfirmation => {
+          dynamicsLib.findById.mockImplementationOnce(() => ({}))
+          dynamicsLib.findByExample.mockImplementationOnce(() => [])
+          const mockPayload = {
+            preferredMethodOfConfirmation: preferredMethodOfConfirmation
+          }
+          const permit = mockPermit()
+          const contact = await resolveContactPayload(permit, mockPayload)
+          expect(contact.shortTermPreferredMethodOfConfirmation.description).toEqual(preferredMethodOfConfirmation)
+        }
+      )
+    })
```

Comment:
Your each cases are all the same, so you could simplify this by doing `describe.each(['Letter', 'Email', 'Text'])(() => {`